packages/compiler-sfc/src/cssVars.ts

```diff
@@ -37,7 +37,10 @@ export function parseCssVars(sfc: SFCDescriptor): string[] {
   sfc.styles.forEach(style => {
     let match
     while ((match = cssVarRE.exec(style.content))) {
-      vars.push(match[1] || match[2] || match[3])
+      const variable = match[1] || match[2] || match[3]
+      if (!vars.includes(variable)) {
```

Comment:
I wonder if we shouldn't just use a Set instead of an array

Reply:
I thought the same but I wanted a minimal PR for this bug fix. The `array.includes` pattern is used in several other places in compiler-sfc also. I saw several places where a Set could be used. In this particular case, I was thinking if creating the extra set will not end up affecting components with a small number of vars (1-4), that should be the common scenario.
But I can change the code to use a Set it if you prefer that.